src/segmentedfilesystem/FileRetriever.java

```diff
@@ -1,25 +1,107 @@
 package segmentedfilesystem;
 
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class FileRetriever {
+        String server;
+        int port;
+        DatagramSocket socket;
+
+        // Constructor
+        public FileRetriever(String server, int port) throws UnknownHostException, SocketException {
+                this.server = server;
+                this.port = port;
+                DatagramSocket socket = new DatagramSocket();
+                this.socket = socket;
+        }
+
+        // method to start conversation with server
+        public static void start(int port, String server, DatagramSocket socket) throws IOException {
+                InetAddress inetAddress = InetAddress.getByName(server);
+                // DatagramSocket socket = new DatagramSocket(port);
+                byte[] buf = new byte[0];
+                DatagramPacket dp = new DatagramPacket(buf, buf.length, inetAddress, port);
+                socket.send(dp);
+        }
+
+        public void getPackets(String server, int port, DatagramSocket socket) throws IOException {
+                byte[] buf = new byte[0];
+                int numFiles = 3; // Three here becasue this is number in writeup
+                // Give 'dummy' values to packet list IDs to signify they are open for use
+                List<PacketManager> packyList = new ArrayList<PacketManager>();
+                for (int i = 0; i < numFiles; i++) {
+                        PacketManager packy = new PacketManager(Byte.MAX_VALUE);
```

Comment:
Why call a constructor with a "fake" value like this? Why not just have the constructor put that value in for you?